src/content/core/okteto-variables.mdx

```diff
@@ -7,36 +7,259 @@ id: okteto-variables
 
-Okteto supports manual definition of variables, offering significant flexibility in configuration. These user-defined variables are separate from Okteto's default, built-in variables. We recommend creating variables in the following order of preference:
+- **Managed by Okteto**: variables injected in your dev environment automatically by Okteto
+- **The `--var` flag**: Used to define variables ad-hoc when using the Okteto CLI
+- **Catalog**: Used to define project-specific variables
+- **Local Environment Variables**: Used to define variables ad-hoc when using the Okteto CLI
+- **The `.env` file**: Used to define project or team-specific variables
+- **User Variables**: Used to define variables at the user level
+- **Admin Variables**: Used to define variables for all users in the organization
 
-- Local variables
-- Deploy time variables
-- Catalog variables
-- User variables (in settings)
-- [Admin variables](admin/dashboard.mdx#admin-variables)
+:::note
+
+The above listing order is also the order in which the Okteto CLI gives precedence to each variable type. At the top the variables with the highest precedence.
+
+Variables with the same name defined in different types will be overridden by the variable with higher precedence.
+
+Example: the var `PORT` defined both in `.env` as `PORT=3000` and as `okteto deploy --var PORT=4000`, the value of `PORT` will be `4000` because the `--var` flag has higher precedence.
+
+:::
+
+
+### Scope
+
+Okteto Variables are scoped to the execution of the Okteto CLI. This means that they are:
+
+- available for expanding values in the Okteto manifest
+- available in the `deploy`, `destroy` and `test` commands in the Okteto manifest
+- available to scripts and tools running as part of `deploy`, `destroy` and `test` commands in the Okteto manifest
+
+They **are not** propagated to the development containers. If you need to pass variables to your development containers, see [Development Containers](/docs/development/containers).
 
 :::note
 
-Variables will be overridden by those above them in this list. For example, local variables will override admin variables.
+Variables defined in a specific command item are only scoped to that item by default. To share values between command items, use the special variable [$OKTETO_ENV](#sharing-variables-between-commands-with-okteto_env).
+
+Example:
+
+```yaml
+deploy:
+  commands:
+    - export MY_VAR=hello
+    - echo $MY_VAR # -> ''
+    - echo "MY_VAR=hello" >> $OKTETO_ENV
+    - echo $MY_VAR # -> 'hello'
+```
+
+:::
+
+## Use Cases
+
+### Expand values in the Okteto manifest
+
+Variables can be used to **expand values** in the the **Okteto manifest** (see `${PORT}`)
+
+```yaml
+# okteto.yml
+dev:
+  api:
+    command: npm start
+    environment:
+      PORT: ${PORT}
+```
+
+### Deploy, destroy and test commands
+
+Variables can be used in **deploy**, **destroy** and **test** commands in the Okteto manifest (see `${ENV}`)
+
+```yaml
```

Comment:
I think this sample is redundant

Reply:
Agree, I want to make sure everything is still mentioned explicitly, that commands and tools/scripts have access to Okteto Variables, but I've condensated the 2 pagragraphs into 1 example.

<img width="1046" alt="Screenshot 2024-05-10 at 16 28 52" src="https://github.com/okteto/docs/assets/2318450/bdc8d102-13cd-4137-a44d-842d64e8fe38">


Happy to iterate further if anyone has feedback 🙏